cmd/kubelet/app/server.go

```diff
@@ -130,6 +131,7 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
 	fs.Var(&s.ClusterDNS, "cluster_dns", "IP address for a cluster DNS server.  If set, kubelet will configure all containers to use this for DNS resolution in addition to the host's DNS servers")
 	fs.BoolVar(&s.ReallyCrashForTesting, "really_crash_for_testing", s.ReallyCrashForTesting, "If true, crash with panics more often.")
 	fs.DurationVar(&s.StreamingConnectionIdleTimeout, "streaming_connection_idle_timeout", 0, "Maximum time a streaming connection can be idle before the connection is automatically closed.  Example: '5m'")
+	fs.StringVar(&s.NetworkPluginName, "network_plugin_name", s.NetworkPluginName, "The name of the network plugin executable which will be called from hook points in kubelet lifecycle")
```

Comment:
It's too generic - you need to know what interfaces a plugin supports.  a
network plugin has different flags than a volume plugin

On Mon, Mar 16, 2015 at 8:58 PM, Vipin Jain <notifications@github.com>
wrote:

> In cmd/kubelet/app/server.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5069#discussion_r26547924>
> :
>
> > @@ -130,6 +131,7 @@ func (s *KubeletServer) AddFlags(fs *pflag.FlagSet) {
> >  	fs.Var(&s.ClusterDNS, "cluster_dns", "IP address for a cluster DNS server.  If set, kubelet will configure all containers to use this for DNS resolution in addition to the host's DNS servers")
> >  	fs.BoolVar(&s.ReallyCrashForTesting, "really_crash_for_testing", s.ReallyCrashForTesting, "If true, crash with panics more often.")
> >  	fs.DurationVar(&s.StreamingConnectionIdleTimeout, "streaming_connection_idle_timeout", 0, "Maximum time a streaming connection can be idle before the connection is automatically closed.  Example: '5m'")
> > +	fs.StringVar(&s.NetworkPluginName, "network_plugin_name", s.NetworkPluginName, "The name of the network plugin executable which will be called from hook points in kubelet lifecycle")
>
> @thockin <https://github.com/thockin> how about '--plugin', because this
> can be used for any plugin, doesn't have to be network. Is it too generic?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5069/files#r26547924>
> .
>